internal/api/products.go

```diff
@@ -2,58 +2,210 @@ package api
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// Only allow GET requests.
-		if r.Method != http.MethodGet {
-			w.WriteHeader(http.StatusMethodNotAllowed)
-			return
-		}
 
-		// Get products and associated data from Algolia
-		products, err := getProductsData(a)
-		if err != nil {
-			log.Error("error getting products from algolia", "error", err)
-			http.Error(w, "Error getting product mappings",
-				http.StatusInternalServerError)
-			return
-		}
+		switch r.Method {
+		case "POST":
+			// Decode request.
+			var req ProductRequest
+			if err := decodeRequest(r, &req); err != nil {
+				log.Error("error decoding products request", "error", err)
+				http.Error(w, fmt.Sprintf("Bad request: %q", err),
+					http.StatusBadRequest)
+				return
+			}
 
-		w.Header().Set("Content-Type", "application/json")
-		w.WriteHeader(http.StatusOK)
+			// Add the data to both algolia and the Postgres Database
+			err := AddNewProducts(ar, aw, db, req)
+			if err != nil {
+				log.Error("error inserting new product/Business Unit", "error", err)
+				http.Error(w, "Error inserting products",
+					http.StatusInternalServerError)
+				return
+			}
 
-		enc := json.NewEncoder(w)
-		err = enc.Encode(products)
-		if err != nil {
-			log.Error("error encoding products response", "error", err)
-			http.Error(w, "Error getting products",
-				http.StatusInternalServerError)
+			// Send success response
+			// Send success response with success message
+			response := struct {
+				Message string `json:"message"`
+			}{
+				Message: "Product/BU Inserted successfully",
+			}
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+			enc := json.NewEncoder(w)
+			err = enc.Encode(response)
+
+		case "GET":
+			// Get products and associated data from Algolia
+			products, err := getProductsData(db)
+			if err != nil {
+				log.Error("error getting products from database", "error", err)
+				http.Error(w, "Error getting product mappings",
+					http.StatusInternalServerError)
+				return
+			}
+
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+
+			enc := json.NewEncoder(w)
+			err = enc.Encode(products)
+			if err != nil {
+				log.Error("error encoding products response", "error", err)
+				http.Error(w, "Error getting products",
+					http.StatusInternalServerError)
+				return
+			}
+		default:
+			w.WriteHeader(http.StatusMethodNotAllowed)
 			return
+
 		}
+
 	})
 }
 
 // getProducts gets the product or area name and their associated
-// data from Algolia
-func getProductsData(a *algolia.Client) (map[string]structs.ProductData, error) {
-	p := structs.Products{
-		ObjectID: "products",
-		Data:     make(map[string]structs.ProductData, 0),
-	}
+// data from Database
+func getProductsData(db *gorm.DB) (map[string]struct {
+	Abbreviation   string      `json:"abbreviation"`
+	PerDocTypeData interface{} `json:"perDocTypeData"`
```

Comment:
"PerDocTypeData" this node name is not revealing the puspose. Please find a better name

Reply:
this attribute is never used and normally initialized to null, so the purpose is not quite clear at this point of time.